### PR TITLE
Shutdown robot if auto-dock fails in go-to-kitchen demo

### PR DESCRIPTION
**Pull request by 708yamaguchi**
Merge after https://github.com/knorth55/jsk_robot/pull/168

@tkmtnt7000
Does this commit seem OK to you?
https://github.com/knorth55/jsk_robot/commit/53a9067f84cf0e23345119f9001b4a11f5087a4d

### CODE REVIEW

**Comment by tkmtnt7000** [comment]
https://github.com/knorth55/jsk_robot/commit/53a9067f84cf0e23345119f9001b4a11f5087a4d
I was thinking about this a bit when I created #166.
It looks good to me from the standpoint of protecting the battery from degradation.
However, I feel that if the shutdown command is called here, app_manager may not send an email with the results.

File: jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l [hunk 1]
```diff
@@ -561,6 +560,7 @@
             (:auto-dock-failure
               '(lambda (userdata)
                  (report-auto-dock-failure)
+                 (ros::service-call "shutdown" (instance std_srvs::EmptyRequest :init))
```
Comment (tkmtnt7000) [inline jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l]: I think `:room-light-off` does not run when auto-dock fails, if shutdown is starting in this position.
But I don't know where to put it...

**Comment by 708yamaguchi** [comment]
Thank you for your review!
I am very very sorry for force-pushing.

I decided to edit `app_publisher`.
With the following PR, `app_publisher` can publish topics depending on app's end condition.
https://github.com/knorth55/app_manager_utils/pull/40

In this PR, fetch will publish `/shutdown` topic only when `go-to-kitchen` demo fails.

I will check if this PR works tomorrow.

**Comment by tkmtnt7000** [comment]
Never mind. 
Thank you very much for coming up with a solution.
> In this PR, fetch will publish /shutdown topic only when go-to-kitchen demo fails.

That's exactly what I've been wanting to do!
It looks good to me.

**Comment by 708yamaguchi** [comment]
I checked that fetch shuts down after timeout of go-to-kitchen demo.

**Pull request merged**